test/extended/util/disruption/controlplane/termination.go

```diff
@@ -0,0 +1,120 @@
+package controlplane
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"k8s.io/apimachinery/pkg/runtime"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"k8s.io/client-go/tools/cache"
+
+	"k8s.io/client-go/kubernetes"
+
+	corev1 "k8s.io/api/core/v1"
+
+	kwatch "k8s.io/apimachinery/pkg/watch"
+
+	"k8s.io/client-go/tools/watch"
+
+	"k8s.io/kubernetes/test/e2e/framework"
+	"k8s.io/kubernetes/test/e2e/upgrades"
+)
+
+// GracefulTerminationTest watches events of non-graceful terminations of kube-apiserver.
+type GracefulTerminationTest struct{}
+
+func (GracefulTerminationTest) Name() string { return "kube-apiserver-terminates-gracefully" }
+func (GracefulTerminationTest) DisplayName() string {
+	return "Kubernetes API server terminates gracefully"
```

Comment:
`[sig-something]` prefix.  Decide if you think it's kube-apiserver or network.

Reply:
Unfortunately this can have many reasons, from ourselves, to node, to cri-o.